ImageInLib/ImageInLib/src/imageInterpolation.h

```diff
@@ -18,6 +18,32 @@ extern "C" {
 
 	bool upSampling(dataType** originalImage, dataType** newImage, size_t length, size_t width, size_t height);
 
+	//=========================
+
+	//Structure to handle image spacing
+	typedef struct {
+		dataType sx, sy, sz;
+	} Spacing3D;
+
+	//Structure to handle image coordinates
+	//We need new structure because coordinates are array indexes (integer type is needed)
+	typedef struct {
+		size_t x, y, z;
+	} imgPoint3D;
+
+	/*
+	* Get real world cordinate from image coordinate
+	* srcPoint : contains the voxel indexes
+	* realOrigin : image origin in real world
+	* imageSpacing : - distance between voxels in x and y direction
+	*                - distance between slices
+	* orientation : IJK(1,1,1), RAS(-1,-1,1) or LPS(1,-1,-1)
+	*/
+	Point3D imageCoordToRealCoord(imgPoint3D srcPoint, Point3D realOrigin, Spacing3D imageSpacing, Point3D orientation);
```

Comment:
I guess, orientation should not be Point3D. It is not enough, because it should represent three 3D vectors. It could be new structure Orientation with three attributes of Point3D

Reply:
I update the function. The first one is ready but I am still working on the second one (from real world coordinates to image coordinates)